include/recursive_graph_bisection.hpp

```diff
@@ -0,0 +1,408 @@
+#pragma once
+
+#include <cmath>
+#include <fstream>
+#include <iterator>
+#include <thread>
+#include <vector>
+
+#include "pstl/algorithm"
+#include "pstl/execution"
+#include "tbb/task_group.h"
+
+#include "forward_index.hpp"
+#include "util/index_build_utils.hpp"
+#include "util/log.hpp"
+#include "util/progress.hpp"
+#include "util/single_init_vector.hpp"
+
+namespace ds2i {
+const Log2<4096> log2;
+
+#ifndef NDEBUG
```

Comment:
This is somewhat temporary. I needed some assertion at hoc, didn't want to put it anywhere else, since I know you've looked into assertions and maybe already planned to do something else.

Reply:
Oh ok, read only now. Just commented about that. Thanks for clarifying